src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.CompilationManager.cs

```diff
@@ -0,0 +1,218 @@
+                var includeSuppressedDiagnostics = true;
+                var newAnalyzerDriverOpt = await CreateAnalyzerDriverAsync(project, stateSets, includeSuppressedDiagnostics, cancellationToken).ConfigureAwait(false);
+
+                // Add new analyzer driver to the map
+                analyzerDriverOpt = _map.GetValue(project, _ => newAnalyzerDriverOpt);
+
+                // if somebody has beat us, make sure analyzers are good.
+                if (analyzerDriverOpt != newAnalyzerDriverOpt)
+                {
+                    AssertAnalyzers(analyzerDriverOpt, stateSets);
+                }
+
+                // return driver
+                return analyzerDriverOpt;
+            }
+
+            public Task<CompilationWithAnalyzers> CreateAnalyzerDriverAsync(Project project, IEnumerable<StateSet> stateSets, bool includeSuppressedDiagnostics, CancellationToken cancellationToken)
+            {
+                var analyzers = stateSets.Select(s => s.Analyzer).ToImmutableArrayOrEmpty();
+                return CreateAnalyzerDriverAsync(project, analyzers, includeSuppressedDiagnostics, cancellationToken);
+            }
+
+            public async Task<CompilationWithAnalyzers> CreateAnalyzerDriverAsync(
+                Project project, ImmutableArray<DiagnosticAnalyzer> analyzers, bool includeSuppressedDiagnostics, CancellationToken cancellationToken)
+            {
+                if (!project.SupportsCompilation)
+                {
+                    return null;
+                }
+
+                var compilation = await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+
+                // Create driver that holds onto compilation and associated analyzers
+                return CreateAnalyzerDriver(
+                    project, compilation, analyzers, logAnalyzerExecutionTime: false, reportSuppressedDiagnostics: includeSuppressedDiagnostics);
+            }
+
+            private CompilationWithAnalyzers CreateAnalyzerDriver(
+                Project project,
+                Compilation compilation,
+                ImmutableArray<DiagnosticAnalyzer> analyzers,
+                bool logAnalyzerExecutionTime,
+                bool reportSuppressedDiagnostics)
+            {
+                // PERF: there is no analyzers for this compilation.
+                //       compilationWithAnalyzer will throw if it is created with no analyzers which is perf optimization.
+                if (analyzers.IsEmpty)
+                {
+                    return null;
+                }
+
+                Contract.ThrowIfFalse(project.SupportsCompilation);
+                AssertCompilation(project, compilation);
+
+                var analysisOptions = GetAnalyzerOptions(project, logAnalyzerExecutionTime, reportSuppressedDiagnostics);
+
+                // Create driver that holds onto compilation and associated analyzers
+                return compilation.WithAnalyzers(analyzers, analysisOptions);
```

Comment:
We should filter out suppressed analyzers like v1 does here: http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs,188
Otherwise, CompilationWithAnalyzers will unnecessary try to execute them, probably causing some extra allocations, and eventually bail out.